e2e/test/MessageSendE2EMultiplexingTests.cs

```diff
@@ -242,12 +236,84 @@ public async Task Message_X509_DeviceSendSingleMessage_MuxWithPooling_AmqpWs()
                 foreach (DeviceClient deviceClient in deviceClients)
                 {
                     await deviceClient.CloseAsync().ConfigureAwait(false);
-                    _log.WriteLine($"{nameof(MessageSendE2ETests)}: Disposing deviceClient {TestLogging.GetHashCode(deviceClient)}");
+
+                    // The connection status change count should be 2: connect (open) and disabled (close)
+                    Assert.IsTrue(deviceClientConnectionStatusChangeCount[deviceClient] == 2);
+
+                    _log.WriteLine($"{nameof(MessageSendE2EMultiplexingTests)}: Disposing deviceClient {TestLogging.GetHashCode(deviceClient)}");
                     deviceClient.Dispose();
                 }
             }
         }
 
+        private async Task SendMessageModuleMuxedOverAmqp(
+            Client.TransportType transport,
+            int poolSize,
+            int devicesCount,
+            bool useSameDevice = false)
+        {
+            var transportSettings = new ITransportSettings[]
+            {
+                new AmqpTransportSettings(transport)
+                {
+                    AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                    {
+                        MaxPoolSize = unchecked((uint)poolSize),
+                        Pooling = true
+                    }
+                }
+            };
+
+            ICollection<ModuleClient> moduleClients = new List<ModuleClient>();
+            Dictionary<ModuleClient, int> moduleClientConnectionStatusChangeCount = new Dictionary<ModuleClient, int>();
+
+            try
+            {
+                _log.WriteLine($"{nameof(MessageSendE2EMultiplexingTests)}: Starting the test execution for {devicesCount} modules");
+
+                for (int i = 0; i < devicesCount; i++)
+                {
+                    ConnectionStatus? lastConnectionStatus = null;
+                    ConnectionStatusChangeReason? lastConnectionStatusChangeReason = null;
+                    int setConnectionStatusChangesHandlerCount = 0;
+
+                    string devicePrefix = useSameDevice ? DevicePrefix : $"{DevicePrefix}_{i}_";
+                    string modulePrefix = useSameDevice ? $"{ModulePrefix}_{i}_" : ModulePrefix;
+
+                    TestModule testModule = await TestModule.GetTestModuleAsync(devicePrefix, modulePrefix).ConfigureAwait(false);
+                    ModuleClient moduleClient = ModuleClient.CreateFromConnectionString(testModule.ConnectionString, transportSettings);
+                    moduleClients.Add(moduleClient);
+
+                    moduleClient.SetConnectionStatusChangesHandler((status, statusChangeReason) =>
+                    {
+                        setConnectionStatusChangesHandlerCount++;
```

Comment:
Same as above - if at all possible, avoid using lambdas with captures where lifetime of the variables is not clear.